### ISSUE TITLE
Algebra: Implementierung der mathematischen Hilfs-Funktionen

### ISSUE THREAD

**Issue by Helco** [Helco/PebbleOfDoom#19]
- Fixed-point arithmetic
- Vector operations
- ~Rotation matrix operation~ postponed
- Missing utility functions (e.g. abs, min, max, clamp, lerp, etc.)
- Add gtest to project and test math functions

Blocked by #10
